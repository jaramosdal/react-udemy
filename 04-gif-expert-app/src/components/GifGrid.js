import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const [iamges, setIamges] = useState([])

    useEffect(() => {
        getGifs(category).then(setIamges);
    }, [category]);

    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
            {
                iamges.map((img) => (
                    <GifGridItem 
                        key={ img.id }
                        { ...img } 
                    />
                ))
            }
            </div>
        </>
       
    )
}
