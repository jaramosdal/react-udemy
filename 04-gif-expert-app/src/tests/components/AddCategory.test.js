import React from 'react';
import { shallow } from 'enzyme';

import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });

    test('debemos mostrar <AddCategory /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe cambiar la caja de texto', () => {
        const input = wrapper.find('input'); 
        const value = 'Hola Mundo';
        input.simulate('change', { 
            target: {
                value
            }
         });
    });

    test('NO debe postear la informaciÃ³n con submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('debe llamar al setCategories y limpiar la caja de texto', () => {
        const value = 'Hola Mundo';
        
        const input = wrapper.find('input');
        input.simulate('change', { target: { value } }); 

        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        
        expect(input.prop('value')).toBe('');
    });
});