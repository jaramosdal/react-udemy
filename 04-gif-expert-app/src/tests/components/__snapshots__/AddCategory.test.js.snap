// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe cambiar la caja de texto 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;

exports[`Pruebas en <AddCategory /> debemos mostrar <AddCategory /> correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;

exports[`Pruebas en <AddCategory /> debemos mostrar <AddCategory /> correctamente 2`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
